torch/_inductor/lowering.py

```diff
@@ -1202,16 +1202,27 @@ def slice_(x, dim=0, start=0, end=2**63, step=1, clamp=True):
 
 @register_lowering(aten.as_strided, type_promotion_kind=None)
 def as_strided(x, size, stride, storage_offset=None):
+    new_device = None
+    new_dtype = None
     if isinstance(x, TensorBox) and isinstance(x.data, ir.BaseView):
-        # as_strided ignores views
+        # Note: Merging views
+        # When we use as_strided, we can rewrite the size/stride/offset
+        # of the incoming buffer x. If x is a view, we would overwrite
+        # its metadata. Except for device type and dtype, which we need
+        # to propagate.
+
+        # Technically device is not needed because it is handled as a copy
+        # instead of a view today.
```

Comment:
Nit: `Except for device type and dtype, which we need
        # to propagate.`

rewrite this ? it's not possible to take cross-device views.